src/pool/srv_pool_map.c

```diff
@@ -116,10 +123,20 @@ update_one_tgt(struct pool_map *map, struct pool_target *target,
 				DP_TARGET(target));
 			return -DER_BUSY;
 		case PO_COMP_ST_DOWNOUT:
+			if (!is_command && target->ta_comp.co_flags & PO_COMPF_COMMAND) {
+				D_DEBUG(DB_MD, DF_TARGET" must be reinted by admin command\n",
+					DP_TARGET(target));
+				return -DER_NO_PERM;
```

Comment:
So if the target is excluded by cmd, then swim can not reint the target?  This is a bit strange. IMHO.  I thought DAOS_SELF_HEAL_AUTO_REINT should control this?  i.e. the user should understand the target will be reintegrated once it is startup.

Hmm, if there are duplicate reintegrations due to this, the 2nd can always be ignored?  Or I miss some cases?

Reply:
We want the disk and the rank  to be marked down by the command must also be marked up by the command.
A rank has multiple disks, when there are bad disks, slow disks and other scenarios, we hope that the target on these faulty disks in the rank does not go up, which can solve this scenario

Reply:
I see. thanks for clarification. Though I wonder

1. should we add an option in the cmd (dmg pool exclude) to indicate these ranks/targets are not allow to be reintegrated automatically (by swim). Or it should be by default. @liw any thought here?
2. pool map version needs to updated for adding new flag?

Reply:
> 1. should we add an option in the cmd (dmg pool exclude) to indicate these ranks/targets are not allow to be reintegrated automatically (by swim). Or it should be by default. @liw any thought here?

Well, I don't have a clear preference, as long as the reintegration process can handle the churn. By the way, thanks for reminding me of this patch; I will revise my #7678 to accommodate the changes here.

Reply:
> 

Flag setting and cancellation are accompanied by target status changes

Reply:
Any suggestions?